### DISCUSSION TITLE
Support of PIL.ImageOps.exif_transpose(image)?

### DISCUSSION THREAD

**Discussion by Soooda** [bigcat88/pillow_heif#85] [Q&A] [answered]
Hello all,

I tried to convert some of my HIF images into JPGs and found some of the images were throwing the below error:
```
ValueError: corrupted image(dimensions in header: (7008, 4672), decoded dimensions: (4672, 7008)
```
The root cause was that I had applied rotation to my images and on the IOS system, the rotation was applied to the EXIF rather than the image data itself.

<img width="314" alt="image" src="https://user-images.githubusercontent.com/27431435/225835035-925fcc36-c840-44f8-bfc0-ae9d10a5cb02.png">

I made use of [`exif_transpose()`](https://pillow.readthedocs.io/en/stable/reference/ImageOps.html#PIL.ImageOps.exif_transpose) to compensate for that. But the error is still popping up. It looks like `pillow_heif` does not support `exif_transpose()` and I wonder if this is an intended behaviour?

```python
from PIL import Image, ImageOps
from pillow_heif import register_heif_opener

register_heif_opener()

def save_jpg(path, quality=100, subsampling=0):
    temp, _ = path.split(".")
    temp += ".jpg"

    im = Image.open(path)
    im = ImageOps.exif_transpose(im)

    print(f"Saving to {temp}")
    im.save(temp, quality=quality, subsampling=subsampling)

# for filename in sys.argv[1:]:
#     save_jpg(filename)

name = "DSC03260.HIF"
save_jpg(name)
```

**Comment by bigcat88** [comment, marked as answer]
@Soooda post a link to an image, for me currently it looks like a bug. Issue #86

**Reply by bigcat88** [reply]
So far, guessing on a cup of coffee instead of an image, I can make an assumption that the application / device that created the image indicated different size headers in the file header and in the image itself. (Yes, the HEIF standard has two different fields for this, and this is without Exif) More info here: https://github.com/strukturag/libheif/issues/784
Specifically, this case can be corrected (in theory), the number of pixels in the total in the image has not changed.
I need to have an example image to tell exactly and publish a fix.

**Reply by Soooda** [reply]
Soz for the late reply, was into something on the weekend xD. Bear with me, I was writing the program to convert my photos out of Sony A7M4 into JPGs. So the file is sorta oversized.

https://drive.google.com/file/d/1oLoAyazNfKww08y6I3psu8vEl4WQ5Kp-/view?usp=sharing
